packages/std/src/results/cosmos_msg.rs

```diff
@@ -59,6 +59,32 @@ pub enum CosmosMsg<T = Empty> {
     Gov(GovMsg),
 }
 
+impl<T> CosmosMsg<T> {
+    /// Convert this this [`CosmosMsg<T>`] to a [`CosmosMsg<U>`] with a different custom message type.
+    /// This allows easier interactions between code written for a specific chain and
+    /// code written for multiple chains.
+    /// If this is the [`CosmosMsg::Custom`] variant, the function returns `None`.
+    pub fn change_custom<U>(self) -> Option<CosmosMsg<U>> {
+        Some(match self {
+            CosmosMsg::Bank(msg) => CosmosMsg::Bank(msg),
+            CosmosMsg::Custom(_) => return None,
```

Comment:
This is a bit of a random idea and I don't know if possible or needed. We should do something like this to make it more powerful:

```suggestion
            CosmosMsg::Custom(msg) => CosmosMsg::Custom(msg.into()),
```

in cases where old and new type have an Into relationship.

But feel free to ignore if not useful.

Reply:
I was wondering about something similar while writing this. Something like a `map_custom(self, f: impl FnOnce(T) -> U) -> CosmosMsg<U>`. But I think its utility is pretty limited: Converting between custom messages of different chains is probably not possible in the general case. Only case that is always possible is custom to empty, but that also doesn't make much sense to me, since it just silently removes the custom message, so I discarded that idea.